splice_machine/src/main/java/com/splicemachine/derby/vti/SchemaFilterVTI.java

```diff
@@ -0,0 +1,95 @@
+import com.splicemachine.db.iapi.sql.conn.Authorizer;
+import com.splicemachine.db.iapi.sql.conn.ConnectionUtil;
+import com.splicemachine.db.iapi.sql.conn.LanguageConnectionContext;
+import com.splicemachine.db.iapi.sql.dictionary.SchemaDescriptor;
+import com.splicemachine.db.iapi.sql.dictionary.StatementPermission;
+import com.splicemachine.db.iapi.sql.dictionary.StatementSchemaPermission;
+import com.splicemachine.db.iapi.sql.execute.ExecRow;
+import com.splicemachine.db.iapi.store.access.TransactionController;
+import com.splicemachine.db.iapi.types.SQLVarchar;
+import com.splicemachine.db.impl.jdbc.EmbedConnection;
+import com.splicemachine.db.impl.jdbc.EmbedResultSetMetaData;
+import com.splicemachine.db.impl.sql.catalog.DataDictionaryImpl;
+import com.splicemachine.db.impl.sql.execute.ValueRow;
+import com.splicemachine.db.vti.VTICosting;
+import com.splicemachine.db.vti.VTIEnvironment;
+import com.splicemachine.derby.iapi.sql.execute.SpliceOperation;
+import com.splicemachine.derby.stream.iapi.DataSet;
+import com.splicemachine.derby.stream.iapi.DataSetProcessor;
+import com.splicemachine.derby.stream.iapi.OperationContext;
+import com.splicemachine.derby.utils.SpliceAdmin;
+import com.splicemachine.derby.vti.iapi.DatasetProvider;
+import com.splicemachine.pipeline.Exceptions;
+import com.splicemachine.si.impl.driver.SIDriver;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SchemaFilterVTI implements DatasetProvider, VTICosting {
+    private OperationContext operationContext;
+
+    @Override
+    public double getEstimatedRowCount(VTIEnvironment vtiEnvironment) throws SQLException {
+        return 1;
+    }
+
+    @Override
+    public double getEstimatedCostPerInstantiation(VTIEnvironment vtiEnvironment) throws SQLException {
+        return 1;
+    }
+
+    @Override
+    public boolean supportsMultipleInstantiations(VTIEnvironment vtiEnvironment) throws SQLException {
+        return false;
+    }
+
+    @Override
+    public DataSet<ExecRow> getDataSet(SpliceOperation op, DataSetProcessor dsp, ExecRow execRow) throws StandardException {
+        operationContext = dsp.createOperationContext(op);
+
+        ArrayList<ExecRow> items = new ArrayList<ExecRow>();
+
+        Activation activation = op.getActivation();
+        LanguageConnectionContext lcc = activation.getLanguageConnectionContext();
+
+        TransactionController tc = lcc.getTransactionExecute();
+        DataDictionaryImpl dd = (DataDictionaryImpl)lcc.getDataDictionary();
+        List<SchemaDescriptor> schemas= dd.getAllSchemas(tc);
+
+        for(SchemaDescriptor schema: schemas) {
+            String schemaName = schema.getSchemaName();
+            StatementPermission statementPermission = new StatementSchemaPermission(schemaName, null, Authorizer.ACCESS_PRIV);
+            if (!lcc.getAuthorizer().canSeeSchema(lcc.getLastActivation(), statementPermission)) {
```

Comment:
Is there a particular reason to use lcc.getLastActivation() instead of the activation variable above?
Also a repeat of the same question from SYSSCHEMASRowFactory: why returning the non-accessible schemas instead of the accessible schemas directly?

Reply:
@yxia92  I missed the activation above. I don't think there is any difference so changed the code.